drivers/dummy/rasterizer_dummy.h

```diff
@@ -725,20 +725,25 @@ class RasterizerStorageDummy : public RasterizerStorage {
 	}
 
 	bool free(RID p_rid) {
-
 		if (texture_owner.owns(p_rid)) {
 			// delete the texture
 			DummyTexture *texture = texture_owner.get(p_rid);
 			texture_owner.free(p_rid);
 			memdelete(texture);
-		}
-
-		if (mesh_owner.owns(p_rid)) {
+		} else if (mesh_owner.owns(p_rid)) {
 			// delete the mesh
 			DummyMesh *mesh = mesh_owner.getornull(p_rid);
 			mesh_owner.free(p_rid);
 			memdelete(mesh);
+		} else if (lightmap_capture_data_owner.owns(p_rid)) {
+			// delete the lightmap
+			LightmapCapture *lightmap_capture = lightmap_capture_data_owner.getornull(p_rid);
+			lightmap_capture_data_owner.free(p_rid);
+			memdelete(lightmap_capture);
+		} else {
+			return false;
 		}
+
 		return true;
```

Comment:
this becomes unreachable right?
Just remove the `else` block and change this to `false` then?

Reply:
It still looks reachable by all the other branches of the `if`?

Reply:
Indeed, it's used whenever something has been freed.

Reply:
Yeah I saw that pattern in the gles3
Class. Personally I'd prefer setting some sort of ret_val type of variable, and then having a single return statement. Is there any sort of standard pattern you guys use for this sort of thing?

Reply:
I think the current pattern is fine.